File: payroll-api/src/main/java/com/company/payroll/mapper/TrainingMapper.java
package com.company.payroll.mapper;

import java.util.List;

import org.springframework.stereotype.Repository;

import com.company.payroll.model.Training;

@Repository
public interface TrainingMapper {
    int deleteByPrimaryKey(Integer tId);

    int insert(Training row);

    int insertSelective(Training row);
    
    List<Training> selectList();
    
    List<Training> selectListByEId(int eid);
    
    List<Training> selectListByMId(int mid);

    Training selectByPrimaryKey(Integer tId);

    int updateByPrimaryKeySelective(Training row);

    int updateByPrimaryKeyWithBLOBs(Training row);

    int updateByPrimaryKey(Training row);
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/mapper/TrainingMapper.kt
package com.company.payroll.mapper

import com.company.payroll.model.Training
import org.springframework.stereotype.Repository

@Repository
interface TrainingMapper {
  fun deleteByPrimaryKey(tId: Int): Int
  fun insert(row: Training): Int
  fun insertSelective(row: Training): Int
  fun selectList(): List<Training>?
  fun selectListByEId(staffId: Int): List<Training>?
  fun selectByPrimaryKey(tId: Int): Training?
  fun updateByPrimaryKeySelective(row: Training): Int
  fun updateByPrimaryKeyWithBLOBs(row: Training): Int
  fun updateByPrimaryKey(row: Training): Int
}